### DISCUSSION TITLE
TeXStudio Scripts

### DISCUSSION THREAD

**Discussion by nsaxena1234** [texstudio-org/texstudio#1987] [General]
Hi Forum,

I am new to TeXStudio and need some help....

I want to run different Perl scripts on the loading and close of m TeX file in TeXstudio. Please help and provide your suggestions....

Best Wishes,
NS

**Comment by muzimuzhi** [comment]
TeXstudio macros' event triggers `?load-file` and `?close-file` and command `system(cmd, workingDirectory="")` may be helpful. See [user manual](https://htmlpreview.github.io/?https://github.com/texstudio-org/texstudio/master/utilities/manual/usermanual_en.html), sec. 4.6.3 "Script Macros" and sec. 4.6.4.3 "Event Triggers".

**Reply by nsaxena1234** [reply]
Thank dear, that is really helpful...I got the option....
